[DEMIS2SORMAS] Create a dedicated menu for messages [3]

<!--
If you've never submitted an issue to the SORMAS repository before or this is your first time using this template, please read the Contributing guidelines (https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/CONTRIBUTING.md) for an explanation of the information we need you to provide. You don't have to remove this comment or any other comment from this issue as they will automatically be hidden.
-->
### Feature Description
<!-- Mandatory -->
Samples > Lab Messages / DEMIS interface
### Problem Description
<!-- Mandatory -->
Thus far, external messages in regards to case surveillance, especially in association with the DEMIS interface, have been managed in the lab messages directory. Recently, more message types have been introduced to DEMIS which requires users in SORMAS to not only receive and process lab messages but also other kinds of messages, e.g. physician reports.
In order to make the management of external messages more generic and clear, the lab messages directory should become its own menu item and serve as a general location to process any type of message that report cases to users.

### Proposed Change
<!-- Mandatory -->

- [x] Move the lab messages directory out of the sample menu and create a separate menu item called 'Messages'. Place this menu item in between Tasks and Persons. Use the notepad icon as seen elsewhere.
    - [x] Rename the directory title to 'Message Directory'
    - [x] Rename 'Fetch lab messages' to 'Fetch messages'
    - [x] Rename 'Lab name' to 'Reporter name' (included in search field placeholder)
    - [x] Rename 'Lab postal code' to 'Reporter postal code' (included in search field placeholder)
    - [x] Rename the pop-up header 'Lab message' to 'Message' (visible when viewing the message details)

- [x] Rename some further strings associated with general messages by replacing 'lab message(s)' with just 'message(s)' in the following contexts:
    - [x] labMessageLabMessagesList
    - [x] confirmationDeleteLabMessages
    - [x] confirmationUnclearLabMessage
    - [x] confirmationManuallyForwardedLabMessage
    - [x] confirmationFetchLabMessages
    - [x] confirmationSinceLabMessages
    - [x] headingLabMessageDownload
    - [x] headingNoLabMessagesSelected
    - [x] headingLabMessagesDeleteProcessed
    - [x] headingLabMessagesDeleted
    - [x] headingFetchLabMessages
    - [x] messageLabMessagesAssigned
    - [x] messageForwardedLabMessageFound
    - [x] messageNoLabMessagesSelected
    - [x] messageLabMessagesDeleteProcessed
    - [x] messageLabMessagesDeleted
    - [x] labMessageAlreadyProcessedError
    - [x] labMessagePdfConversionError

- [x] Rename the feature configurations `LAB_MESSAGES` to `EXTERNAL_MESSAGES` and adjust related descriptions
- [x] Change the `LAB_MESSAGES` user right to become generic MESSAGES rights
    - [x] Introduce the user right `EXTERNAL_MESSAGES_VIEW` - users with this right should be able to view the Messages directory, message details of the contained messages and download messages
    - [x] Introduce the user right `EXTERNAL_MESSAGE_PROCESS` - users with this right should be able to fetch messages, edit the assignee and status of a message as well as process them
    - [x] Introduce the user right `EXTERNAL_MESSAGE_DELETE` user right - users with this right should be able to delete messages
    - [x] Only users should be able to be set as assignees that also have the `EXTERNAL_MESSAGE_PROCESS`



### Possible Alternatives
<!-- Optional -->

### Additional Information
<!-- Optional -->
![Message menu](https://user-images.githubusercontent.com/70317594/164263618-6c425d93-1b98-4628-8477-56980dd4ee26.png)


![Message Directory](https://user-images.githubusercontent.com/70317594/164257948-82461d1a-70fc-42bc-85d1-51fc24622905.png)

This issue seems to break the build in Jenkins:

Error Message
invalid versioning trigger externalmessage  DELETE 
Stacktrace
java.lang.AssertionError: 

invalid versioning trigger externalmessage  DELETE 
	at de.symeda.sormas.backend.common.StartupShutdownServiceTest.testHistoryTablesMatch(StartupShutdownServiceTest.java:147)

--- a/sormas-ui/src/main/java/de/symeda/sormas/ui/externalmessage/ExternalMessageGridFilterForm.java
+++ b/sormas-ui/src/main/java/de/symeda/sormas/ui/externalmessage/ExternalMessageGridFilterForm.java
@@ -73,7 +73,7 @@ public class ExternalMessageGridFilterForm extends AbstractFilterForm<ExternalMe
 		searchField.setNullRepresentation("");
 
 		ComboBoxWithPlaceholder assignee = addField(ExternalMessageCriteria.ASSIGNEE, ComboBoxWithPlaceholder.class);
-		assignee.addItem(new UserReferenceDto(ReferenceDto.NO_REFERENCE_UUID, I18nProperties.getCaption(Captions.unassigned), "", null));
+		assignee.addItem(new UserReferenceDto(ReferenceDto.NO_REFERENCE_UUID, "", "", I18nProperties.getCaption(Captions.unassigned)));
 		assignee.addItems(FacadeProvider.getUserFacade().getUsersByRegionAndRights(user.getRegion(), null, UserRight.EXTERNAL_MESSAGE_PROCESS));
 		assignee.setNullSelectionAllowed(true);
 

